Entities/AtomGroup.cpp

```diff
@@ -1,2618 +1,1750 @@
+		RTEAssert(m_OwnerMO, "Traveling an AtomGroup without a parent MO!");
+
+		bool didWrap = false;
+		Vector pushImpulse;
+
+		// Pin the path to where the owner wants the joint to be.
+		limbPath.SetJointPos(jointPos);
+		limbPath.SetJointVel(velocity);
+		limbPath.SetRotation(rotation);
+		limbPath.SetFrameTime(travelTime);
+
+		const Vector distVec = g_SceneMan.ShortestDistance(jointPos, m_LimbPos);
+		// Restart the path if the limb is way off somewhere else
+		if (distVec.GetMagnitude() > m_OwnerMO->GetDiameter()) { limbPath.Terminate(); }
+
+		do {
+			if (limbPath.PathEnded()) {
+				if (restarted) { *restarted = true; }
+				if (!limbPath.RestartFree(m_LimbPos, m_OwnerMO->GetRootID(), m_OwnerMO->IgnoresWhichTeam())) {
+					return false;
+				}
+			}
+			// Do the push travel calculations and get the resulting push impulse vector back.
+			pushImpulse = PushTravel(m_LimbPos, limbPath.GetCurrentVel(m_LimbPos), limbPath.GetPushForce(), didWrap, limbPath.GetNextTimeChunk(m_LimbPos), false, false);
+
+			// Report back to the path where we've ended up.
+			limbPath.ReportProgress(m_LimbPos);
+
+		// End the path push loop if the path has ended or we ran out of time.
+		} while (!limbPath.FrameDone() && !limbPath.PathEnded());
+
+		// Sanity check this force coming out of really old crummy physics code
+		if (pushImpulse.GetLargest() > 10000.0F) { pushImpulse.Reset(); }
+
+		// Add the resulting impulse force, add the lever of the joint offset if set to do so
+		m_OwnerMO->AddImpulseForce(pushImpulse, affectRotation ? (g_SceneMan.ShortestDistance(m_OwnerMO->GetPos(), jointPos) * c_MPP) : Vector());
+
+		return true;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void AtomGroup::FlailAsLimb(const Vector &ownerPos, const Vector &jointOffset, const float limbRadius, const Vector &velocity, const float angularVel, const float mass, const float travelTime) {
+		RTEAssert(m_OwnerMO, "Traveling an AtomGroup without a parent MO!");
+
+		bool didWrap = false;
+		Vector jointPos = ownerPos + jointOffset;
+		Vector centrifugalVel = jointOffset * std::fabs(angularVel);
+
+		// Do the push travel calculations and get the resulting push impulse vector back.
+		Vector pushImpulse = PushTravel(m_LimbPos, velocity + centrifugalVel, 100, didWrap, travelTime, false, false, false);
+
+		// Constrain within the range of the limb
+		Vector limbVec = m_LimbPos - jointPos;
+
+		if (limbVec.GetMagnitude() > limbRadius) {
+			limbVec.SetMagnitude(limbRadius);
+			m_LimbPos = jointPos + limbVec;
 		}
-/*
+		return;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool AtomGroup::InTerrain() const {
+		RTEAssert(m_OwnerMO, "Using an AtomGroup without a parent MO!");
+
+		if (!g_SceneMan.SceneIsLocked()) { g_SceneMan.LockScene(); }
```

Comment:
Locking doesn't really do anything right now, honestly i'm not even sure why it's there since it's only relevant for video bitmaps.
I guess I should comment all of these out and keep them for future reference.

Reply:
It seems like junk code here to me? Doesn't do much, isn't actually respected cause even if it fails to lock the code does its thing, and it's not in the other similar method. I'd kill it off if it's pointless, though obviously keep the SceneMan method cause it's valuable.

Reply:
Still gonna keep it here as reference, just commented out.

Reply:
¯\_(ツ)_/¯
Fine, I'm not really a fan of that but I'll live with it, sure.